src/parsers/perf/perfparser.cpp

```diff
@@ -1009,6 +1012,38 @@ class PerfParserPrivate : public QObject
         }
     }
 
+    void addTracepointToTracepointStack(uint32_t tid, const Data::Tracepoint& tracepoint)
+    {
+        const QVector<Data::TracepointTimeMeasurementsParameters> tracepoints =
+            Settings::instance()->tracepointParameters();
+
+        auto addToLookup = [this, &tracepoints](const QString& name) -> bool {
+            for (const auto& parameters : tracepoints) {
+                auto match = parameters.startRegex.match(name);
+                if (match.hasMatch()) {
+                    QString stopName = parameters.stopRegex.pattern();
```

Comment:
huh what is this doing here? why do you take the pattern and make it a string? this looks very odd to me - don't you instead just want to store the stop regex directly and use that later on? or match the stopregex directly and replace some of its matches somehow? but taking the pattern and doing stuff with its string sounds totally backwards - why is it a regex then to begin with?

Reply:
I tried implementing regex support as requested in #346. To improve performance I wanted to cache the regexes. It seems  I didn't notice that parameters.stopRegex doesn't need to be a regex